src/core/pointcloud/qgscopcpointcloudindex.cpp

```diff
@@ -43,6 +43,35 @@ QgsCopcPointCloudIndex::QgsCopcPointCloudIndex() = default;
 
 QgsCopcPointCloudIndex::~QgsCopcPointCloudIndex() = default;
 
+QgsPointCloudIndex *QgsCopcPointCloudIndex::clone() const
+{
+  QgsCopcPointCloudIndex *clone = new QgsCopcPointCloudIndex;
+  QMutexLocker locker( &mHierarchyMutex );
+
+  // Base QgsPointCloudIndex fields
+  clone->mExtent = mExtent;
```

Comment:
These should be moved to a protect "copycommo properties" method in the base class to avoid the duplication